html/modules/custom/hr_paragraphs/src/Controller/HdxController.php

```diff
@@ -0,0 +1,313 @@
+            'url' => Url::fromUserInput($base_url, [
+              'query' => [
+                'filters' => array_merge_recursive($filters, $filter),
+              ],
+            ]),
+          ];
+        }
+
+        if (count($links) > 1) {
+          $facet_blocks[$name] = [
+            'title' => $this->getHdxFilters($name),
+            'links' => $links,
+          ];
+        }
+      }
+    }
+
+    return $facet_blocks;
+  }
+
+  /**
+   * Build Hdx parameters.
+   */
+  public function buildHdxParameters(int $offset, int $limit, array $query_filters) : array {
+    $parameters = [
+      'q' => '',
+      'fq' => '+dataset_type:dataset -extras_archived:true',
+      'fq_list' => [],
+      'facet.field' => [
+        'groups',
+        'res_format',
+        'organization',
+        'vocab_Topics',
+        'license_id',
+      ],
+      'facet.limit' => 250,
+      'sort' => 'score desc, if(gt(last_modified,review_date),last_modified,review_date) desc',
+      'start' => $offset,
+      'rows' => $limit,
+    ];
+
+    // Pasted filters from URL are using OR.
+    foreach ($query_filters as $key => $values) {
+      switch ($key) {
+        case 'q':
+        case 'sort':
+          $parameters[$key] = $values;
+          break;
+
+        case 'groups':
+        case 'res_format':
+        case 'organization':
+        case 'vocab_Topics':
+        case 'license_id':
+          if (is_array($values)) {
+            $parameters['fq_list'][] = $key . ':"' . implode('" OR "', $values) . '"';
+          }
+          else {
+            $parameters['fq_list'][] = $key . ':"' . $values . '"';
+          }
+          break;
+
+      }
+    }
+
+    return $parameters;
+  }
+
+  /**
+   * Allowed filters.
+   */
+  public function getHdxFilters($key = NULL) {
+    $filters = [
+      'groups' => $this->t('Groups'),
+      'res_format' => $this->t('Formats'),
+      'organization' => $this->t('Organizations'),
+      'vocab_Topics' => $this->t('Tags'),
+      'license_id' => $this->t('Licenses'),
+    ];
+
+    if ($key) {
+      if (array_key_exists($key, $filters)) {
+        return $filters[$key];
+      }
+      else {
+        return FALSE;
+      }
+    }
+    else {
+      return $filters;
+    }
+  }
+
+  /**
+   * Execute Hdx query.
+   */
+  public function executeHdxQuery(array $parameters) : array {
+    $endpoint = 'https://data.humdata.org/api/3/action/package_search';
+
+    try {
+      $response = $this->httpClient->request(
+        'POST',
+        $endpoint,
+        [
+          RequestOptions::JSON => $parameters,
+        ]
+      );
+    }
+    catch (RequestException $exception) {
+      if ($exception->getCode() === 404) {
+        throw new NotFoundHttpException();
+      }
+    }
+
+    $body = $response->getBody() . '';
+    $results = json_decode($body, TRUE);
+
+    return $results['result'];
+  }
+
+  /**
+   * Build Hdx objects.
+   */
+  public function buildHdxObjects(array $results) : array {
```

Comment:
@rupl list of available fields in twig

Reply:
Nice it looks like I can probably cherry-pick the templates on top of this. We picked the same twig variable names in our two branches.